libp2p/NodeTable.h

```diff
@@ -114,10 +114,20 @@ class NodeTable : UDPSocketEvents
     /// Returns distance based on xor metric two node ids. Used by NodeEntry and NodeTable.
     static int distance(NodeID const& _a, NodeID const& _b) { u256 d = sha3(_a) ^ sha3(_b); unsigned ret; for (ret = 0; d >>= 1; ++ret) {}; return ret; }
 
+    /// Open the udp socket and schedule execution of the discovery algorithm
+    void start();
```

Comment:
Why is this needed, can't we do it in constructor as earlier?

Reply:
We can’t call shared_from_this inside of the ctor